schema_editor/app/scripts/views/recordtype/preview-controller.js

```diff
@@ -0,0 +1,58 @@
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function RTPreviewController($stateParams, RecordTypes, RecordSchemas) {
+        var ctl = this;
+        ctl.onDataChange = angular.noop;
+        initialize();
+
+        function initialize() {
+            loadRecordType()
+                .then(loadRecordSchema)
+                .then(onSchemaReady);
+        }
+
+        // Helper for loading the record type
+        function loadRecordType () {
+            return RecordTypes.get({ id: $stateParams.uuid })
+                .$promise.then(function(recordType) {
+                    ctl.recordType = recordType;
+                });
+        }
+
+        // Helper for loading the record schema
+        function loadRecordSchema() {
+            /* jshint camelcase: false */
+            var currentSchemaId = ctl.recordType.current_schema;
+            /* jshint camelcase: true */
+
+            return RecordSchemas.get({ id: currentSchemaId })
+                .$promise.then(function(recordSchema) {
+                    ctl.recordSchema = recordSchema;
+                });
+        }
+
+        // Called after all prerequesite data has been loaded
+        function onSchemaReady() {
+            ctl.editor = {
+                id: 'schema-editor',
```

Comment:
This should probably be different since this is a preview of a data entry form rather than a schema editing form.